src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -685,6 +675,52 @@ private void getCanvasResponse(HttpServletRequest request,
 		out.flush();
 	}
 
+	//When there is too much data to retrieve at once data will be paged.
+	//When data is paged Link Headers will be send in http response. We are
+	//only concerned with the Next Link Header. This method will extract header 
+	//if one exists.
+	private String extractNextLink(HttpResponse canvasResponse) {
+		String linkValueString = null;
+		String linkValueNext = null;
+		String searchPhrase = "rel=\"next\"";
+		M_log.debug("SEARCH PHRASE: " + searchPhrase);
+		HashMap<String, String> links = new HashMap<String, String>();
+		Header[] headers = canvasResponse.getAllHeaders();
+		for (Header header : headers) {
+			M_log.debug("Key : " + header.getName() 
+					+ " ,Value : " + header.getValue());
+			if(header.getName().equals("Link")){
+				linkValueString = header.getValue();
+				String[] headerPairs = linkValueString.split(",");
+				for(String headerPair : headerPairs){
+					M_log.debug("String: " + headerPair);
+					String[] splitPairs = headerPair.split(";");
```

Comment:
If there's a *simple* regular expression for this splitting and the following replacements (and possibly trimming), use that instead.